### PR TITLE
Using The RESTEasy Tracing Feature In WildFly

### PR DESCRIPTION
**Pull request by liweinan**

### CODE REVIEW

**Comment by liweinan** [comment]
@jamezp @rsearls This is still a draft, and I'll finish writing this blog post this week. Because there are already blog posts and document describing the tracing feature itself, so the focus is on the tracing feature usage in WildFly. If you have any suggestions on this article please comment and I'll do the modifications :D

File: _posts/2023-04-12-tracing-feature-in-wildfly.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+title:      "Using The RESTEasy Tracing Feature In WildFly"
+subtitle:   ""
+date:       2023-04-12
+author:     Wei Nan Li
+---
+
+
+The RESTEasy tracing feature has been integrated into WildFly since its version 28. The usage of the tracing feature in RESTEasy is already introduced in RESTEasy Documentation[^resteasydoc] and some blog posts[^resteasyposts]. In this article, I’ll focus on the usage of this feature in WildFly.
+
+There is an example in the `resteasy-example`[^resteaasyexample] project showing the usage of the tracing feature, and it uses the Galleon Feature Pack to provision a WildFly 28 server, so I’ll use this example this this article.
```
Comment (jamezp) [inline _posts/2023-04-12-tracing-feature-in-wildfly.md:12]: I would say "it uses Galleon" here.

**Review by jamezp — COMMENTED** [review]
This is just a general comment about the example itself and something I think I'll look at when I make a few small changes to how the example provisions a server.

I wonder if we should create a `header` resource which would show how the headers could be used too. This would show either a JSON response or plain text for the header so the user doesn't have to leave the browser to see the logs. I do think keeping this blog post focused on the logs is fine though.

**Comment by rsearls** [comment]
With those minor changes it looks fine.

File: _posts/2023-04-12-tracing-feature-in-wildfly.md (hunk 1 above)
Comment (liweinan) [reply _posts/2023-04-12-tracing-feature-in-wildfly.md:12]: okay!

File: _posts/2023-04-12-tracing-feature-in-wildfly.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+And then entering the `tracing-example` directory and run the provisioned WildFly server:
+
+```bash
+$ pwd
+/Users/weli/works/resteasy-examples/tracing-example
+```
+
+```bash
+$ mvn wildfly:run
```
[the remark this answers is not in the document]
Comment (liweinan) [reply _posts/2023-04-12-tracing-feature-in-wildfly.md:30]: yes sir!

File: _posts/2023-04-12-tracing-feature-in-wildfly.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+If we need to change the configuration, we need to change the above change and redeploy the project. On the other hand, in WildFly 28, because the tracing feature is already integrated, so we can use the WildFly Command Line Interface to configure the tracing feature at runtime. Using the `jboss-cli.sh` in the provisioned WildFly server to connect to the WildFly server:
+
+```bash
+$ pwd
+/Users/weli/works/resteasy-examples/tracing-example/target/wildfly/bin
+```
+
+```bash
+➤ ./jboss-cli.sh
```
[the remark this answers is not in the document]
Comment (liweinan) [reply _posts/2023-04-12-tracing-feature-in-wildfly.md:78]: cool!

File: _posts/2023-04-12-tracing-feature-in-wildfly.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+```bash
+tracing-threshold=SUMMARY
+tracing-type=OFF
+```
+
+Now we can change the `tracing-type` from `OFF` to `ON` on the fly:
+
+```bash
+[standalone@localhost:9990 /] cd subsystem=jaxrs
```
[the remark this answers is not in the document]
Comment (liweinan) [reply _posts/2023-04-12-tracing-feature-in-wildfly.md:111]: I'll modify it :D

File: _posts/2023-04-12-tracing-feature-in-wildfly.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+```bash
+"process-state" => "reload-required"
+```
+
+So we need to use the following command to reload the WildFly configuration:
+
+```bash
+[standalone@localhost:9990 subsystem=jaxrs] cd /
+[standalone@localhost:9990 /] :reload
```
[the remark this answers is not in the document]
Comment (liweinan) [reply _posts/2023-04-12-tracing-feature-in-wildfly.md:139]: okay!

**Comment by liweinan** [comment]
> This is just a general comment about the example itself and something I think I'll look at when I make a few small changes to how the example provisions a server.
> 
> I wonder if we should create a `header` resource which would show how the headers could be used too.

Could you please help to provide this in the `resteasy-example` side so I could catch your thought? And I'll update this blog post accordingly :D

> This would show either a JSON response or plain text for the header so the user doesn't have to leave the browser to see the logs. I do think keeping this blog post focused on the logs is fine though.

**Comment by liweinan** [comment]
@jamezp @rsearls Thanks for reviewing this post! I'll update the text accordingly.

**Comment by liweinan** [comment]
> > This is just a general comment about the example itself and something I think I'll look at when I make a few small changes to how the example provisions a server.

> > I wonder if we should create a `header` resource which would show how the headers could be used too.
> 
> @jamezp Could you please help to provide this in the `resteasy-example` side so I could catch your thought? And I'll update this blog post accordingly :D

Guess I understand your meaning now. I'll provide a header resource in the example :D
> 
> > This would show either a JSON response or plain text for the header so the user doesn't have to leave the browser to see the logs. I do think keeping this blog post focused on the logs is fine though.

**Comment by liweinan** [comment]
Relative PR: https://github.com/resteasy/resteasy-examples/pull/123

**Comment by liweinan** [comment]
@jamezp I have updated the blog post, and create a PR to the `tracing-example` to add notes about JSON format header. I moved this PR to `ready for review`, could you please help to check if anything else need to be modified? Thanks!

File: _posts/2023-04-12-tracing-feature-in-wildfly.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+
+[^resteasydoc]: [Chapter 54. RESTEasy Tracing Feature](https://docs.jboss.org/resteasy/docs/6.2.3.Final/userguide/html_single/index.html#Tracing_Feature)
+
+[^resteasyposts]: [A brief introduction to the RESTEasy Tracing Feature](https://resteasy.dev/2018/06/11/a-brief-introduction-to-the-resteasy-tracing-feature/) / [RESTEasy Tracing Feature Now - Supports JSON formatted information](https://resteasy.dev/2018/09/05/resteasy-tracing-feature-now-supports-json-formatted-information/)
+
+[^resteaasyexample]: https://github.com/resteasy/resteasy-examples/pull/122
+
+[^galleon]: [Galleon Provisioning](https://github.com/wildfly/galleon)
+
+[^wildflytracing]: [tracingEnabled in WildFly](https://github.com/wildfly/wildfly/pull/15198/files#diff-6716f11308b916188cda71be70b388833fb56713df3644b1f4a41fa4815c8275R230)
```
Comment (jamezp) [inline _posts/2023-04-12-tracing-feature-in-wildfly.md:204]: I wonder if this should link the documentation itself https://github.com/wildfly/wildfly/pull/15198/files#diff-b1b3a12a4b68c3397dacaab84d50790141d87677f46552c5138f520312aa3782R155.

File: _posts/2023-04-12-tracing-feature-in-wildfly.md [hunk 7]
```diff
@@ -0,0 +1,205 @@
+---
+layout:     post
+title:      "Using The RESTEasy Tracing Feature In WildFly"
+subtitle:   ""
+date:       2023-04-12
```
Comment (jamezp) [inline _posts/2023-04-12-tracing-feature-in-wildfly.md:5]: We should change this and the file name IMO before we merge this.

File: _posts/2023-04-12-tracing-feature-in-wildfly.md [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+First to build the project and provision the WildFly server:
+
+```bash
+$ mvn install
+```
+
+The above command will build the example, download necessary WildFly components. Then run the provisioned WildFly server with the following command:
+
+```bash
+$ mvn wildfly:dev
```
Comment (jamezp) [inline _posts/2023-04-12-tracing-feature-in-wildfly.md:38]: You might want to use `mvn wildfly:provision wildfly:dev` here. If a previous `install` was not done, the `jboss-home` won't be found.

File: _posts/2023-04-12-tracing-feature-in-wildfly.md [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+} 
+```
+
+From the above command output, it says:
+
+```bash
+"process-state" => "reload-required"
+```
+
+So we need to use the following command to reload the WildFly configuration:
```
Comment (jamezp) [inline _posts/2023-04-12-tracing-feature-in-wildfly.md:133]: "So we need to use the following command to reload the WildFly services:". "configuration" doesn't sound right here to me.

File: _posts/2023-04-12-tracing-feature-in-wildfly.md [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+
+From the above output, we can see the tracing feature is disabled.
+
+## References
+
+[^resteasydoc]: [Chapter 54. RESTEasy Tracing Feature](https://docs.jboss.org/resteasy/docs/6.2.3.Final/userguide/html_single/index.html#Tracing_Feature)
+
+[^resteasyposts]: [A brief introduction to the RESTEasy Tracing Feature](https://resteasy.dev/2018/06/11/a-brief-introduction-to-the-resteasy-tracing-feature/) / [RESTEasy Tracing Feature Now - Supports JSON formatted information](https://resteasy.dev/2018/09/05/resteasy-tracing-feature-now-supports-json-formatted-information/)
+
+[^resteaasyexample]: https://github.com/resteasy/resteasy-examples/pull/122
```
Comment (jamezp) [inline _posts/2023-04-12-tracing-feature-in-wildfly.md:200]: Shouldn't this just point to https://github.com/resteasy/resteasy-examples/tree/main/tracing-example

**Comment by liweinan** [comment]
@jamezp thanks for the review! I'll finish writing this blog post by this week.

**Comment by liweinan** [comment]
@jamezp I have finished updating the text, could you please help to do a review? Thanks!

**Pull request merged**